skt/executable.py

```diff
@@ -460,6 +461,50 @@ def cmd_report(cfg):
     reporter.report()
 
 
+def cmd_console_check(cfg):
+    """
+    Check the console logs for any traces.
+
+    Args:
+        cfg: A dictionary of skt configuration.
+    """
+    remove_oldresult(cfg.get('output_dir'), 'console_check.')
+    console_result_path = join_with_slash(cfg.get('output_dir'),
+                                          'console_check.result')
+    console_report_path = join_with_slash(cfg.get('output_dir'),
+                                          'console_check.report')
+
+    if not cfg.get('krelease') or not cfg.get('console'):
+        raise Exception('<krelease> or <console-url> parameter missing!')
+
+    console_logs = []
+    trace_list = []
+
+    for console_path_or_url in cfg.get('console'):
+        console_log = skt.console.ConsoleLog(cfg.get('krelease'),
+                                             console_path_or_url)
+
+        console_logs.append(console_log)
+        trace_list.append(console_log.gettraces())
+
+    if any(trace_list):
+        report_string = ''
+
+        for traces in trace_list:
+            if traces:
```

Comment:
Why do we check this?

Reply:
Because the method for extracting the traces returns an empty list if there are no traces found, and we don't want to say "we found an empty trace"

Reply:
I see, thank you.

This is one of the situations, I think, where complex logical checks Python calls for are making for a difficult read. I think an explicit check for list emptiness/length would have been better, but once you make it clear that you're operating on a list (e.g. by renaming `traces` to `trace_list` or similar) it will be easier to understand.